test/e2e/subscription_e2e_test.go

```diff
@@ -1366,6 +1366,258 @@ var _ = Describe("Subscription", func() {
+				catsrcDepRight.Spec.Priority = 100
+
+				catsrcDepWrong, c3 = createInternalCatalogSource(kubeClient, crClient, "catsrc2", testNamespace,
+					[]registry.PackageManifest{packageDepWrong}, []apiextensions.CustomResourceDefinition{crd}, csvsWrong)
+				catsrcDepWrong.Spec.Priority = 100
+
+				cleanup = func() {
+					c1()
+					c2()
+					c3()
+				}
+			})
+
+			By("waiting for catalogsources to be ready", func() {
+				_, err := fetchCatalogSourceOnStatus(crClient, catsrcMain.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+				_, err = fetchCatalogSourceOnStatus(crClient, catsrcDepRight.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+				_, err = fetchCatalogSourceOnStatus(crClient, catsrcDepWrong.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("choose the dependent package from the same catsrc that has a higher priority value", func() {
+				ensureInstalledCSVs()
+			})
+		})
+
+		It("the same dependency exists on both the catsrc of the installing operator and another catsrc with a higher"+
+			" priority", func() {
+			packageDepRight = registry.PackageManifest{PackageName: "PackageDependent"}
+			csv := newCSV(rightCSVName, testNamespace, "", semver.MustParse("0.1.0"),
+				[]apiextensions.CustomResourceDefinition{crd}, nil, nil)
+			packageDepRight.DefaultChannelName = alphaChannel
+			packageDepRight.Channels = []registry.PackageChannel{{Name: alphaChannel, CurrentCSVName: csv.GetName()}}
+			csvsMain = append(csvsMain, csv)
+
+			csv.Name = wrongCSVName
+			packageDepWrong = packageDepRight
+			packageDepWrong.Channels = []registry.PackageChannel{{Name: alphaChannel, CurrentCSVName: csv.GetName()}}
+			csvsWrong = append(csvsWrong, csv)
+
+			By("creating catalogsources", func() {
+				var c1, c2 cleanupFunc
+				catsrcMain, c1 = createInternalCatalogSource(kubeClient, crClient, genName("catsrc"), testNamespace,
+					[]registry.PackageManifest{packageDepRight, packageMain}, []apiextensions.CustomResourceDefinition{crd}, csvsMain)
+
+				catsrcDepWrong, c2 = createInternalCatalogSource(kubeClient, crClient, genName("catsrc"), testNamespace,
+					[]registry.PackageManifest{packageDepWrong}, []apiextensions.CustomResourceDefinition{crd}, csvsWrong)
+				catsrcDepWrong.Spec.Priority = 100
+
+				cleanup = func() {
+					c1()
+					c2()
+				}
+			})
+
+			By("waiting for catalogsources to be ready", func() {
+				_, err := fetchCatalogSourceOnStatus(crClient, catsrcMain.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+				_, err = fetchCatalogSourceOnStatus(crClient, catsrcDepWrong.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			By("choose the dependent package from the same catsrc as the installing operator", func() {
```

Comment:
The "By" descriptions in each of these tests are more like what I would expect the "It" descriptions to say -- they are describing the expected behavior.

Reply:
they are more like steps within the It, i dont think they should be it within an it.

Reply:
The string passed to `It` is essentially the "Then" in the BDD "GivenWhenThen", and it represents the behavior that the program being tested is expected to demonstrate. If one of these tests fails before it reaches the end, the test report will not tell you which behavior was broken.

Reply:
so you are suggesting to convert all the By within the It to be It?

Reply:
we can do that

Reply:
> so you are suggesting to convert all the By within the It to be It?

No, only to change the descriptions being passed to the existing `It`s